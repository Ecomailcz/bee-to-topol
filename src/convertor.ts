import {TopolBlock, TopolColumn, TopolSection, TopolTemplate} from "./TopolTemplateTypes";
import {BeeTemplate, Row, Column, WebFont, Module} from "./BeeTemplateTypes";
import {v4 as uuidv4} from 'uuid';
import {convertBlock} from "./blockConvertor";
import {stringWithPxToNumeric, convertFonts} from "./helpers";

type ConvertorWarningMessage = {
    message: "unknown-module-type";
    type: string;
}

type Transformer = {
    key: string;
    convert: (block: Module, columnWidth: number, uuid: string) => TopolBlock;
}

type convertorOptions = {
    convertors?: Transformer[];
}

const warnings: ConvertorWarningMessage[] = []

export default (beeTemplate: BeeTemplate, options?: convertorOptions) => {

    const TEMPLATE_WIDTH = stringWithPxToNumeric(beeTemplate.page.body.content.computedStyle.messageWidth)

    const convertColumn = (column: Column, columnWidth: number, columnWidthPercentage: string): TopolColumn => {

        const blocks: TopolBlock[] = [];

        column.modules.forEach(module => {

            const { block, notConvertedBlock} = convertBlock(module, columnWidth);

            if (notConvertedBlock !== undefined) {
                if(options !== undefined && options.convertors !== undefined) {
                    options.convertors.forEach(convertor => {
                        if(module.type === convertor.key) {
                            const transformedBlock = convertor.convert(notConvertedBlock, columnWidth, uuidv4());
                            blocks.push(transformedBlock);
                        }
                    })
                } else {
                    warnings.push({
                        message: "unknown-module-type",
                        type: module.type
                    })
                }

                return;
            }

            if(block !== undefined) {
                blocks.push(block);
            }
        });
        
        function getPadding() {
            let leftPadding = '0px';
            let rightPadding = '0px';
            let topPadding = '0px';
            let bottomPadding = '0px';

            if(column.style["padding-bottom"]) {
                bottomPadding = column.style["padding-bottom"];
            }
            if(column.style["padding-left"]) {
                leftPadding = column.style["padding-left"];
            }
            if(column.style["padding-right"]) {
                rightPadding = column.style["padding-right"];
            }
            if(column.style["padding-top"]) {
                topPadding = column.style["padding-top"];
            }

            return `${topPadding} ${rightPadding} ${bottomPadding} ${leftPadding}`;
        }

        return {
            tagName: "mj-column",
            attributes: {
                width: columnWidthPercentage,
                "vertical-align": "top",
                "background-color": column.style["background-color"],
                "border-bottom": column.style["border-bottom"],
                "border-top": column.style["border-top"],
                "border-left": column.style["border-left"],
                "border-right": column.style["border-right"],
                "padding": getPadding(),
            },
            children: blocks,
            uid: uuidv4()
        }
    }

    const convertSection = (row: Row): TopolSection => {
        const columns: TopolColumn[] = [];

        if (row.columns.length === 0) {
            throw new Error("Row must have at least one column, invalid template syntax.");
        }

        const columnWidth = TEMPLATE_WIDTH / row.columns.length;

        const columnWidthPercentage = 100 / row.columns.length + "%";

        row.columns.forEach(column => {
            columns.push(convertColumn(column, columnWidth, columnWidthPercentage));
        });

        const getCssClass = () => {
            if (!row.content.computedStyle) {
                return undefined;
            }

            if (row.content.computedStyle.hideContentOnDesktop) {
                return "hide_section_on_desktop";
            }
            if (row.content.computedStyle.hideContentOnMobile) {
                return "hide_section_on_mobile";
            }  

            return undefined;
        }

        const getBackgroundUrl = () => {

            if(row.content.style["background-image"]){
                const rowStyleBgImage = row.content.style["background-image"]
                .replace("url('", "")
                .replace("')", "");

                if (rowStyleBgImage !== "none") {
                    return rowStyleBgImage;
                }   
            }

            if(!row.container.style["background-image"]){
                return 'none';
            }

            return row.container.style["background-image"]
                .replace("url('", "")
                .replace("')", "");

        }

        const getBackgroundColor = () => {
            const rowStyleBgColor = row.content.style["background-color"];

            if (rowStyleBgColor !== "transparent") {
                return rowStyleBgColor;
            }

            return row.container.style["background-color"];
        }

        return {
            tagName: "mj-section",
            attributes: {
                "full-width": false,
                "background-color": getBackgroundColor(),
                //this will be always 0px, padding is set only on columns
                padding: "0px 0px 0px 0px", 
                "css-class": getCssClass(),
                "background-url": getBackgroundUrl(),
                // "background-position": row.content.style["background-position"], //TBD
                "background-size": row.content.style["background-size"],
                "background-repeat": row.content.style["background-repeat"],
                "in-group": row.content.computedStyle && !row.content.computedStyle.rowColStackOnMobile
            },
            children: columns,
            layout: 1,
            backgroundColor: null,
            backgroundImage: null,
            paddingTop: 0,
            paddingBottom: 0,
            paddingLeft: 0,
            paddingRight: 0,
            uid: uuidv4(),
        }
    }

    const convertSections = (rows: Row[]) => {
        const sections: TopolSection[] = [];

        rows.forEach(row => {
            sections.push(convertSection(row));
        })

        return sections;
    }


    const getBgColor = () => {
        if (beeTemplate.page.body.content.computedStyle.messageBackgroundColor !== "transparent") {
            return beeTemplate.page.body.content.computedStyle.messageBackgroundColor;
        }

        if (beeTemplate.page.body.container.style["background-color"]) {
            return beeTemplate.page.body.container.style["background-color"];
        }

        return "transparent";
    }

    const topolTemplate: TopolTemplate = {
        tagName: "mj-global-style",
        attributes: {
            "h1:color": "#000",
            "h1:font-family": "Helvetica, sans-serif",
            "h2:color": "#000",
            "h2:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
            "h3:color": "#000",
            "h3:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
            ":color": beeTemplate.page.body.content.style.color,
            ":font-family": beeTemplate.page.body.content.style["font-family"],
            ":line-height": "1.5",
            "a:color": beeTemplate.page.body.content.computedStyle.linkColor,
            "button:background-color": "#e85034",
            "containerWidth": TEMPLATE_WIDTH,
            "fonts": beeTemplate.page.body.content.style["font-family"],
            "mj-text": {
                "line-height": 1.5,
                "font-size": 15
            },
            "mj-button": [],
        },
        children: [
            {
                tagName: "mj-body",
                attributes: {
                    "background-color": getBgColor(),
                    "containerWidth": stringWithPxToNumeric(beeTemplate.page.body.content.computedStyle.messageWidth)
                },
                children: convertSections(beeTemplate.page.rows),
            }
        ],

        "style": {
            "h1": {
                "font-family": "\"Cabin\", sans-serif",
                "font-size": "22px"
            },
            "h2": {
                "font-size": "17px",
                "font-family": "Ubuntu, Helvetica, Arial"
            },
            "h3": {
                "font-size": "13px",
                "font-family": "Ubuntu, Helvetica, Arial"
            },
            "p": {
                "font-size": "11px",
                "font-family": "Ubuntu, Helvetica, Arial"
            },
            "ul": {
                "font-size": "11px",
                "font-family": "Ubuntu, Helvetica, Arial"
            },
            "li": {
                "font-size": "11px",
                "font-family": "Ubuntu, Helvetica, Arial"
            },
            "ol": {
                "font-size": "11px",
                "font-family": "Ubuntu, Helvetica, Arial"
            },
            "a": {
                "color": beeTemplate.page.body.content.computedStyle.linkColor
            }
        },
        fonts: convertFonts(beeTemplate.page.body.webFonts),
    }

    return {
        template: topolTemplate,
        warnings: warnings,
    }
}