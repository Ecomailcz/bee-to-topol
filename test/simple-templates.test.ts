import {expect, test, vi} from "vitest";
import convertor from "../src/convertor";
import simpleTemplate1 from '../src/bee-templates/simple-templates/simple-template.json';
import simpleTemplateWithButton from '../src/bee-templates/simple-templates/simple-template-with-button.json';
import simpleTemplateWithTwoButtons from '../src/bee-templates/simple-templates/simple-template-with-two-buttons.json';

vi.mock('uuid', () => {
    return {
        v4: () => 'mocked-uuid'
    }
});

test('match snapshot for simple template 1', () => {
    const converted = convertor(simpleTemplate1);
    expect(converted.template).toMatchSnapshot();
});

test('match snapshot for simple template with button', () => {
    // @ts-expect-error
    const converted = convertor(simpleTemplateWithButton);
    expect(converted.template).toMatchSnapshot();
});

test('match snapshot for simple template with two buttons', () => {
    // @ts-expect-error
    const converted = convertor(simpleTemplateWithTwoButtons);
    expect(converted.template).toMatchSnapshot();
});
