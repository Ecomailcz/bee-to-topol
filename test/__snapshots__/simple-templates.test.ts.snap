// Vitest Snapshot v1

exports[`match snapshot for simple template 1 1`] = `
{
  "attributes": {
    ":color": "#000000",
    ":font-family": "TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif",
    ":line-height": "1.5",
    "a:color": "#296eab",
    "button:background-color": "#e85034",
    "containerWidth": 645,
    "fonts": "TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif",
    "h1:color": "#000",
    "h1:font-family": "Helvetica, sans-serif",
    "h2:color": "#000",
    "h2:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
    "h3:color": "#000",
    "h3:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
    "mj-button": [],
    "mj-text": {
      "font-size": 15,
      "line-height": 1.5,
    },
  },
  "children": [
    {
      "attributes": {
        "background-color": "#6f2e2e",
        "containerWidth": 645,
      },
      "children": [
        {
          "attributes": {
            "background-color": "#6f2e2e",
            "background-repeat": "no-repeat",
            "background-size": undefined,
            "background-url": "none",
            "css-class": undefined,
            "full-width": false,
            "in-group": false,
            "padding": "0px 0px 0px 0px",
          },
          "backgroundColor": null,
          "backgroundImage": null,
          "children": [
            {
              "attributes": {
                "background-color": "transparent",
                "border-bottom": "0px solid transparent",
                "border-left": "0px solid transparent",
                "border-right": "0px solid transparent",
                "border-top": "0px solid transparent",
                "padding": "5px 0px 5px 0px",
                "vertical-align": "top",
                "width": "100%",
              },
              "children": [],
              "tagName": "mj-column",
              "uid": "mocked-uuid",
            },
          ],
          "layout": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "tagName": "mj-section",
          "uid": "mocked-uuid",
        },
      ],
      "tagName": "mj-body",
    },
  ],
  "fonts": [
    "'Roboto', Tahoma, Verdana, Segoe, sans-serif",
    "'Abril Fatface', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Merriwheater', 'Georgia', serif",
    "'Montserrat', 'Trebuchet MS', 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', Tahoma, sans-serif",
    "'Nunito', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Bitter', Georgia, Times, 'Times New Roman', serif",
    "'Permanent Marker', Impact, Charcoal, sans-serif",
    "'Roboto Slab', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Cabin', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Oxygen', 'Trebuchet MS', 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', Tahoma, sans-serif",
    "'Oswald', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Lato', Tahoma, Verdana, Segoe, sans-serif",
    "'Ubuntu', Tahoma, Verdana, Segoe, sans-serif",
    "'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
    "'Droid Serif', Georgia, Times, 'Times New Roman', serif",
    "'Playfair Display', Georgia, serif",
    "'Poppins', Arial, Helvetica, sans-serif",
  ],
  "style": {
    "a": {
      "color": "#296eab",
    },
    "h1": {
      "font-family": "\\"Cabin\\", sans-serif",
      "font-size": "22px",
    },
    "h2": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "17px",
    },
    "h3": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "13px",
    },
    "li": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "ol": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "p": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "ul": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
  },
  "tagName": "mj-global-style",
}
`;

exports[`match snapshot for simple template with button 1`] = `
{
  "attributes": {
    ":color": "#000000",
    ":font-family": "TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif",
    ":line-height": "1.5",
    "a:color": "#296eab",
    "button:background-color": "#e85034",
    "containerWidth": 645,
    "fonts": "TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif",
    "h1:color": "#000",
    "h1:font-family": "Helvetica, sans-serif",
    "h2:color": "#000",
    "h2:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
    "h3:color": "#000",
    "h3:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
    "mj-button": [],
    "mj-text": {
      "font-size": 15,
      "line-height": 1.5,
    },
  },
  "children": [
    {
      "attributes": {
        "background-color": "#6f2e2e",
        "containerWidth": 645,
      },
      "children": [
        {
          "attributes": {
            "background-color": "#6f2e2e",
            "background-repeat": "no-repeat",
            "background-size": undefined,
            "background-url": "none",
            "css-class": undefined,
            "full-width": false,
            "in-group": false,
            "padding": "0px 0px 0px 0px",
          },
          "backgroundColor": null,
          "backgroundImage": null,
          "children": [
            {
              "attributes": {
                "background-color": "transparent",
                "border-bottom": "0px solid transparent",
                "border-left": "0px solid transparent",
                "border-right": "0px solid transparent",
                "border-top": "0px solid transparent",
                "padding": "5px 0px 5px 0px",
                "vertical-align": "top",
                "width": "100%",
              },
              "children": [
                {
                  "attributes": {
                    "align": "center",
                    "background-color": "#8a3c90",
                    "border": "0px solid transparent",
                    "border-radius": "4px",
                    "color": "#ffffff",
                    "containerWidth": 645,
                    "css-class": undefined,
                    "font-family": "Ubuntu, Helvetica, Arial, sans-serif, Helvetica, Arial, sans-serif",
                    "font-size": "16px",
                    "href": "",
                    "inner-padding": "5px 20px 5px 20px",
                    "padding": "0px 0px 0px 0px",
                  },
                  "content": "<div class=\\"txtTinyMce-wrapper\\"><p style='font-family: TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif;'>Button</p></div>",
                  "tagName": "mj-button",
                  "uid": "mocked-uuid",
                },
              ],
              "tagName": "mj-column",
              "uid": "mocked-uuid",
            },
          ],
          "layout": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "tagName": "mj-section",
          "uid": "mocked-uuid",
        },
        {
          "attributes": {
            "background-color": "#6f2e2e",
            "background-repeat": "no-repeat",
            "background-size": undefined,
            "background-url": "none",
            "css-class": undefined,
            "full-width": false,
            "in-group": false,
            "padding": "0px 0px 0px 0px",
          },
          "backgroundColor": null,
          "backgroundImage": null,
          "children": [
            {
              "attributes": {
                "background-color": "transparent",
                "border-bottom": "0px solid transparent",
                "border-left": "0px solid transparent",
                "border-right": "0px solid transparent",
                "border-top": "0px solid transparent",
                "padding": "5px 0px 5px 0px",
                "vertical-align": "top",
                "width": "100%",
              },
              "children": [],
              "tagName": "mj-column",
              "uid": "mocked-uuid",
            },
          ],
          "layout": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "tagName": "mj-section",
          "uid": "mocked-uuid",
        },
      ],
      "tagName": "mj-body",
    },
  ],
  "fonts": [
    "'Roboto', Tahoma, Verdana, Segoe, sans-serif",
    "'Abril Fatface', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Merriwheater', 'Georgia', serif",
    "'Montserrat', 'Trebuchet MS', 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', Tahoma, sans-serif",
    "'Nunito', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Bitter', Georgia, Times, 'Times New Roman', serif",
    "'Permanent Marker', Impact, Charcoal, sans-serif",
    "'Roboto Slab', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Cabin', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Oxygen', 'Trebuchet MS', 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', Tahoma, sans-serif",
    "'Oswald', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Lato', Tahoma, Verdana, Segoe, sans-serif",
    "'Ubuntu', Tahoma, Verdana, Segoe, sans-serif",
    "'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
    "'Droid Serif', Georgia, Times, 'Times New Roman', serif",
    "'Playfair Display', Georgia, serif",
    "'Poppins', Arial, Helvetica, sans-serif",
  ],
  "style": {
    "a": {
      "color": "#296eab",
    },
    "h1": {
      "font-family": "\\"Cabin\\", sans-serif",
      "font-size": "22px",
    },
    "h2": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "17px",
    },
    "h3": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "13px",
    },
    "li": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "ol": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "p": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "ul": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
  },
  "tagName": "mj-global-style",
}
`;

exports[`match snapshot for simple template with two buttons 1`] = `
{
  "attributes": {
    ":color": "#000000",
    ":font-family": "TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif",
    ":line-height": "1.5",
    "a:color": "#296eab",
    "button:background-color": "#e85034",
    "containerWidth": 645,
    "fonts": "TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif",
    "h1:color": "#000",
    "h1:font-family": "Helvetica, sans-serif",
    "h2:color": "#000",
    "h2:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
    "h3:color": "#000",
    "h3:font-family": "Ubuntu, Helvetica, Arial, sans-serif",
    "mj-button": [],
    "mj-text": {
      "font-size": 15,
      "line-height": 1.5,
    },
  },
  "children": [
    {
      "attributes": {
        "background-color": "#6f2e2e",
        "containerWidth": 645,
      },
      "children": [
        {
          "attributes": {
            "background-color": "#6f2e2e",
            "background-repeat": "no-repeat",
            "background-size": undefined,
            "background-url": "none",
            "css-class": undefined,
            "full-width": false,
            "in-group": false,
            "padding": "0px 0px 0px 0px",
          },
          "backgroundColor": null,
          "backgroundImage": null,
          "children": [
            {
              "attributes": {
                "background-color": "transparent",
                "border-bottom": "0px solid transparent",
                "border-left": "0px solid transparent",
                "border-right": "0px solid transparent",
                "border-top": "0px solid transparent",
                "padding": "5px 0px 5px 0px",
                "vertical-align": "top",
                "width": "100%",
              },
              "children": [
                {
                  "attributes": {
                    "align": "center",
                    "background-color": "#8a3c90",
                    "border": "0px solid transparent",
                    "border-radius": "4px",
                    "color": "#ffffff",
                    "containerWidth": 645,
                    "css-class": undefined,
                    "font-family": "Ubuntu, Helvetica, Arial, sans-serif, Helvetica, Arial, sans-serif",
                    "font-size": "16px",
                    "href": "",
                    "inner-padding": "5px 20px 5px 20px",
                    "padding": "0px 0px 0px 0px",
                  },
                  "content": "<div class=\\"txtTinyMce-wrapper\\"><p style='font-family: TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif;'>Button</p></div>",
                  "tagName": "mj-button",
                  "uid": "mocked-uuid",
                },
                {
                  "attributes": {
                    "align": "center",
                    "background-color": "#8a3c90",
                    "border": "0px solid transparent",
                    "border-radius": "4px",
                    "color": "#ffffff",
                    "containerWidth": 645,
                    "css-class": undefined,
                    "font-family": "Ubuntu, Helvetica, Arial, sans-serif, Helvetica, Arial, sans-serif",
                    "font-size": "16px",
                    "href": "",
                    "inner-padding": "5px 20px 5px 20px",
                    "padding": "0px 0px 0px 0px",
                  },
                  "content": "<div class=\\"txtTinyMce-wrapper\\"><p style='font-family: TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif;'>Button</p></div>",
                  "tagName": "mj-button",
                  "uid": "mocked-uuid",
                },
                {
                  "attributes": {
                    "align": "center",
                    "background-color": "#8a3c90",
                    "border": "0px solid transparent",
                    "border-radius": "4px",
                    "color": "#ffffff",
                    "containerWidth": 645,
                    "css-class": undefined,
                    "font-family": "Ubuntu, Helvetica, Arial, sans-serif, Helvetica, Arial, sans-serif",
                    "font-size": "16px",
                    "href": "",
                    "inner-padding": "5px 20px 5px 20px",
                    "padding": "0px 0px 0px 0px",
                  },
                  "content": "<div class=\\"txtTinyMce-wrapper\\"><p style='font-family: TimesNewRoman, Times New Roman, Times, Beskerville, Georgia, serif;'>Button</p></div>",
                  "tagName": "mj-button",
                  "uid": "mocked-uuid",
                },
              ],
              "tagName": "mj-column",
              "uid": "mocked-uuid",
            },
          ],
          "layout": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "tagName": "mj-section",
          "uid": "mocked-uuid",
        },
        {
          "attributes": {
            "background-color": "#6f2e2e",
            "background-repeat": "no-repeat",
            "background-size": undefined,
            "background-url": "none",
            "css-class": undefined,
            "full-width": false,
            "in-group": false,
            "padding": "0px 0px 0px 0px",
          },
          "backgroundColor": null,
          "backgroundImage": null,
          "children": [
            {
              "attributes": {
                "background-color": "transparent",
                "border-bottom": "0px solid transparent",
                "border-left": "0px solid transparent",
                "border-right": "0px solid transparent",
                "border-top": "0px solid transparent",
                "padding": "5px 0px 5px 0px",
                "vertical-align": "top",
                "width": "100%",
              },
              "children": [],
              "tagName": "mj-column",
              "uid": "mocked-uuid",
            },
          ],
          "layout": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "tagName": "mj-section",
          "uid": "mocked-uuid",
        },
      ],
      "tagName": "mj-body",
    },
  ],
  "fonts": [
    "'Roboto', Tahoma, Verdana, Segoe, sans-serif",
    "'Abril Fatface', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Merriwheater', 'Georgia', serif",
    "'Montserrat', 'Trebuchet MS', 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', Tahoma, sans-serif",
    "'Nunito', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Bitter', Georgia, Times, 'Times New Roman', serif",
    "'Permanent Marker', Impact, Charcoal, sans-serif",
    "'Roboto Slab', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Cabin', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Oxygen', 'Trebuchet MS', 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', Tahoma, sans-serif",
    "'Oswald', Arial, 'Helvetica Neue', Helvetica, sans-serif",
    "'Lato', Tahoma, Verdana, Segoe, sans-serif",
    "'Ubuntu', Tahoma, Verdana, Segoe, sans-serif",
    "'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
    "'Droid Serif', Georgia, Times, 'Times New Roman', serif",
    "'Playfair Display', Georgia, serif",
    "'Poppins', Arial, Helvetica, sans-serif",
  ],
  "style": {
    "a": {
      "color": "#296eab",
    },
    "h1": {
      "font-family": "\\"Cabin\\", sans-serif",
      "font-size": "22px",
    },
    "h2": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "17px",
    },
    "h3": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "13px",
    },
    "li": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "ol": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "p": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
    "ul": {
      "font-family": "Ubuntu, Helvetica, Arial",
      "font-size": "11px",
    },
  },
  "tagName": "mj-global-style",
}
`;
